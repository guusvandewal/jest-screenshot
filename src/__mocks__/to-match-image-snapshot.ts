export const toMatchImageSnapshot = jest.fn(() => ({ pass: true }));
