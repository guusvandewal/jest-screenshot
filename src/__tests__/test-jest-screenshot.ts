jest.mock("../to-match-image-snapshot");
import { setupJestScreenshot } from "..";
import { toMatchImageSnapshot } from "../to-match-image-snapshot";

describe("jestScreenshot", () => {
    const someBuffer = Buffer.alloc(10);

    it("uses `0` as relative threshold if no threshold is provided", () => {
        setupJestScreenshot();
        expect(someBuffer).toMatchImageSnapshot();
        expect(toMatchImageSnapshot).toHaveBeenCalledWith(
            someBuffer,
            { pixelThresholdRelative: 0 },
            undefined,
        );
    });

    it("throws an error if called outside of jest", () => {
        const originalExpect = expect;
        (global as any).expect = undefined;
        originalExpect(() => setupJestScreenshot()).toThrowErrorMatchingSnapshot();
        (global as any).expect = originalExpect;
    });
});
