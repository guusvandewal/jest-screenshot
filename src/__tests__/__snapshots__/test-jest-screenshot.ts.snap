// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jestScreenshot throws an error if called outside of jest 1`] = `"Jest: Could not find \`expect\`. Can't setup jest-screenshot."`;
